src/detectors/B0ECAL/ProtoCluster_factory_B0ECalTruthProtoClusters.h

```diff
@@ -45,16 +45,15 @@ class ProtoCluster_factory_B0ECalTruthProtoClusters : public eicrecon::JFactoryP
     //------------------------------------------
     // Process
     void Process(const std::shared_ptr<const JEvent> &event) override{
-        // Prefill inputs
-        m_inputHits = event->Get<edm4eic::CalorimeterHit>(m_inputHit_tag);
-        m_mcHits = event->Get<edm4hep::SimCalorimeterHit>(m_inputMCHit_tag);
+        // Get input collection
+        auto hits_coll = static_cast<const edm4eic::CalorimeterHitCollection*>(event->GetCollectionBase(m_inputHit_tag));
+        auto sim_coll = static_cast<const edm4hep::SimCalorimeterHitCollection*>(event->GetCollectionBase(m_inputMCHit_tag));
```

Comment:
I imagine that clang-tidy will suggest `const auto*` for these throughout (readability).

Reply:
Addressed in your PR